stdlib/private/SwiftReflectionTest/SwiftReflectionTest.swift

```diff
@@ -470,8 +476,10 @@ public func reflect(function: @escaping (Int, String, AnyObject?) -> Void) {
       capacity: MemoryLayout<ThickFunction3>.size)
   fn.initialize(to: ThickFunction3(function: function))
 
-  let parts = unsafeBitCast(fn, to: UnsafePointer<ThickFunctionParts>.self)
-  let contextPointer = unsafeBitCast(parts.pointee.context, to: UInt.self)
+  let contextPointer = fn.withMemoryRebound(
+    to: ThickFunctionParts.self, capacity: 1) {
+      UInt(bitPattern: $0.pointee.context)
+  }
```

Comment:
Thanks, @jckarter!

Tried this:

```swift
  let fn = UnsafeMutableRawPointer.allocate(
    bytes: MemoryLayout<ThickFunction3>.size,
    alignedTo: MemoryLayout<ThickFunction3>.alignment)
  fn.initializeMemory(as: ThickFunction3.self,
    to: ThickFunction3(function: function))
  let contextPointer = fn.load(
    fromByteOffset: MemoryLayout<UnsafeRawPointer>.size,
    as: UInt.self)

  reflect(instanceAddress: contextPointer, kind: .Object)

  fn.deallocate(
    bytes: MemoryLayout<ThickFunction3>.size,
    alignedTo: MemoryLayout<ThickFunction3>.alignment)
```

My main concern here is that it is a little less explicit than the typed version. I mean, `ThickFunctionParts` explicitly shows the underlying structure of a function pointer, and the raw pointer version simply uses an offset, which should probably require a comment.

Unless there are other concerns, I would prefer the typed version. What do you think?

Also, the untyped version is longer ;-)